### PR TITLE
feat: make slashing fraction configurable

### PR DESCRIPTION
**Pull request by samigabor**
Add slashingFraction to StreamrConfig
Change math for setting the flag stake
Update commnens in contracts and tests

### CODE REVIEW

File: packages/network-contracts/contracts/OperatorTokenomics/SponsorshipPolicies/VoteKickPolicy.sol [hunk 1]
```diff
@@ -73,21 +73,21 @@ contract VoteKickPolicy is IKickPolicy, Sponsorship {
...
 
         // cache these just in case the config changes during the flag
         flagStakeWei[target] = streamrConfig.flagStakeWei();
         reviewerRewardWei[target] = streamrConfig.flagReviewerRewardWei();
         flaggerRewardWei[target] = streamrConfig.flaggerRewardWei();
 
-        // TODO: after taking at least minimumStake, all 9/10s here are probably overthinking; it's enough that flagStakeWei is 10/9 of the total reviewer reward
-        //       the limit for flagging is 9/10s of the stake so that there's still room to get flagged and lose the remaining 10% of minimumStake
-        // TODO: try to find the "extreme case" by writing more tests and see if the 10/9 can safely be removed
+        // TODO: after taking at least minimumStake, all "1 - slashingFraction" here are probably overthinking
```
Comment (jtakalai) [inline packages/network-contracts/contracts/OperatorTokenomics/SponsorshipPolicies/VoteKickPolicy.sol:86]: I think you can just remove this TODO and simplify the below check

File: packages/network-contracts/contracts/OperatorTokenomics/SponsorshipPolicies/VoteKickPolicy.sol [hunk 2]
```diff
@@ -73,21 +73,21 @@ contract VoteKickPolicy is IKickPolicy, Sponsorship {
...
+        //       the limit for flagging is "1 - slashingFraction" of the stake so that there's still room to get flagged and lose the remaining slashingFraction of minimumStake
+        // TODO: try to find the "extreme case" by writing more tests and see if the check can safely be removed
         committedStakeWei[flagger] += flagStakeWei[target];
-        require(committedStakeWei[flagger] * 10 <= 9 * stakedWei[flagger], "error_notEnoughStake");
+        require(committedStakeWei[flagger]  <= stakedWei[flagger] * (1 ether - streamrConfig.slashingFraction()) / 1 ether, "error_notEnoughStake");
```
Comment (jtakalai) [inline packages/network-contracts/contracts/OperatorTokenomics/SponsorshipPolicies/VoteKickPolicy.sol:90]: simplify this to just `require(committedStakeWei[flagger] <= stakedWei[flagger], "error_notEnoughStake");` unless we can figure out why we need the complexity

**Review by jtakalai — APPROVED** [review]

**Review by jtakalai — CHANGES_REQUESTED** [review]
I actually did request the above changes 😅

File: packages/network-contracts/contracts/OperatorTokenomics/SponsorshipPolicies/VoteKickPolicy.sol (hunk 1 above)
Comment (samigabor) [reply packages/network-contracts/contracts/OperatorTokenomics/SponsorshipPolicies/VoteKickPolicy.sol]: Removed

**Review by jtakalai — APPROVED** [review]

File: packages/network-contracts/contracts/OperatorTokenomics/SponsorshipPolicies/VoteKickPolicy.sol [hunk 3]
```diff
@@ -73,21 +73,18 @@ contract VoteKickPolicy is IKickPolicy, Sponsorship {
...
         flagStakeWei[target] = streamrConfig.flagStakeWei();
         reviewerRewardWei[target] = streamrConfig.flagReviewerRewardWei();
         flaggerRewardWei[target] = streamrConfig.flaggerRewardWei();
 
-        // TODO: after taking at least minimumStake, all 9/10s here are probably overthinking; it's enough that flagStakeWei is 10/9 of the total reviewer reward
-        //       the limit for flagging is 9/10s of the stake so that there's still room to get flagged and lose the remaining 10% of minimumStake
-        // TODO: try to find the "extreme case" by writing more tests and see if the 10/9 can safely be removed
         committedStakeWei[flagger] += flagStakeWei[target];
-        require(committedStakeWei[flagger] * 10 <= 9 * stakedWei[flagger], "error_notEnoughStake");
+        require(committedStakeWei[flagger] <= stakedWei[flagger], "error_notEnoughStake");
```
Comment (samt1803) [inline packages/network-contracts/contracts/OperatorTokenomics/SponsorshipPolicies/VoteKickPolicy.sol:87]: add a test that test this edge case

**Review by samt1803 — CHANGES_REQUESTED** [review]

File: packages/network-contracts/contracts/OperatorTokenomics/SponsorshipPolicies/VoteKickPolicy.sol (hunk 3 above)
Comment (samigabor) [reply packages/network-contracts/contracts/OperatorTokenomics/SponsorshipPolicies/VoteKickPolicy.sol]: Made require more restrictive (it now can flag up to commitedStake - slashingAmount)

**Comment by samigabor** [comment]
Also changed this:
```
function minimumStakeWei() public view returns (uint) {
        return (flaggerRewardWei + flagReviewerCount * flagReviewerRewardWei) * 1 ether / slashingFraction;
    }
```
to this:
```
function minimumStakeWei() public view returns (uint) {
        return (flaggerRewardWei + flagReviewerCount * flagReviewerRewardWei) * 100 * slashingFraction / 1 ether;
    }
```

In the first case the minimumStake would decrease if the slashing increases. Probably not the desired behaviour

**Review by jtakalai — APPROVED** [review]
